File: apps/web/lib/domain/school-day.ts
export const SCHOOL_DAY_START_TIME = "8:00";
export const MORNING_PERIOD_LIMIT = 6;
export const MIN_LUNCH_BREAK_MINUTES = 50;
export const FRIDAY_DAY_INDEX = 4;
export const TEACHER_BREAK_PERIODS = [3, 4, 5] as const;
export const HISTORY_SUBJECT_CODE = "DEJ";

export function schoolPeriodLabel(period: number): string {
  const ordinal = `${period + 1}. hodina`;
  return period === 0 ? `${ordinal} · ${SCHOOL_DAY_START_TIME}` : ordinal;
}

export function crossesLunchBreak(period: number, duration: number): boolean {
  return (
    period < MORNING_PERIOD_LIMIT && period + duration > MORNING_PERIOD_LIMIT
  );
}

export function isAfternoonPeriod(period: number): boolean {
  return period >= MORNING_PERIOD_LIMIT;
}

export function isForbiddenFridayLesson(
  day: number,
  period: number,
  duration: number,
): boolean {
  return day === FRIDAY_DAY_INDEX && period + duration > MORNING_PERIOD_LIMIT;
}

File: apps/solver/app/school_day.py
MORNING_PERIOD_LIMIT = 6
MIN_LUNCH_BREAK_MINUTES = 50
FRIDAY_DAY_INDEX = 4
TEACHER_BREAK_PERIODS = (3, 4, 5)  # zero-based: 4th, 5th, 6th lessons
HISTORY_SUBJECT_CODE = "DEJ"


def crosses_lunch_break(period: int, duration: int) -> bool:
    """Return True when a lesson block would span morning and afternoon teaching."""
    return period < MORNING_PERIOD_LIMIT < period + duration


def is_afternoon_period(period: int) -> bool:
    return period >= MORNING_PERIOD_LIMIT


def is_forbidden_friday_lesson(day: int, period: int, duration: int) -> bool:
    """Friday teaching must finish by the end of the sixth lesson."""
    return day == FRIDAY_DAY_INDEX and period + duration > MORNING_PERIOD_LIMIT
